receiver/awscontainerinsightreceiver/internal/stores/podstore.go

```diff
@@ -229,7 +238,15 @@ func (p *PodStore) setCachedEntry(podKey string, entry *cachedEntry) {
 }
 
 func (p *PodStore) refresh(now time.Time) {
-	podList, _ := p.podClient.ListPods()
+	var podList []corev1.Pod
+	var err error
+	doRefresh := func() {
+		podList, err = p.podClient.ListPods()
+		if err != nil {
+			p.logger.Error("fail to get pod from kubelet", zap.Error(err))
+		}
+	}
+	refreshWithTimeout(p.ctx, doRefresh, refreshInterval)
```

Comment:
What is the timeout on this `Context`?  If it is stored on the `PodStore` struct and not refreshed is it likely to be already expired or so long as to be ineffective?

Reply:
There is no timeout. The context is defined in https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/da6ba4f5515a959883ab34430e2341893b52143a/receiver/awscontainerinsightreceiver/receiver.go#L69.

To make sure the context never timeout or cancel by itself, I changed it to use a background context
```
ctx, acir.cancel = context.WithCancel(context.Background())
```